kratos/utilities/timer.cpp

```diff
@@ -0,0 +1,122 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Pooyan Dadvand
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/timer.h"
+#include "input_output/logger.h"
+
+namespace Kratos
+{
+/// Default constructor.
+Timer::Timer(){}
+
+void Timer::Start(std::string const& rIntervalName)
+{
+    msTimeTable[rIntervalName].SetStartTime(GetTime());
+}
+
+void Timer::Stop(std::string const& rIntervalName)
+{
+    const double stop_time = GetTime();
+    ContainerType::iterator it_time_data = msTimeTable.find(rIntervalName);
+
+    if(it_time_data == msTimeTable.end())
+        return;
+
+    it_time_data->second.Update(stop_time);
+
+    PrintIntervalInformation(rIntervalName, it_time_data->second.GetStartTime(), stop_time);
+}
+
+int Timer::SetOuputFile(std::string const& rOutputFileName)
+{
+    if(msOutputFile.is_open())
+        msOutputFile.close();
+
+    msOutputFile.open(rOutputFileName.c_str());
+
+    msOutputFile << "                                         Start   \tStop     \tElapsed " << std::endl;
+
+    return msOutputFile.is_open();
+}
+
+int Timer::CloseOuputFile()
+{
+    if(msOutputFile.is_open())
+        msOutputFile.close();
+
+    return msOutputFile.is_open();
+}
+
+bool Timer::GetPrintOnScreen()
+{
+    return msPrintOnScreen;
+}
+
+void Timer::SetPrintOnScreen(bool const PrintOnScreen)
+{
+    msPrintOnScreen = PrintOnScreen;
+}
+
+void Timer::PrintIntervalInformation(std::string const& rIntervalName, const double StartTime, const double StopTime)
```

Comment:
Right now each interval timing is always printed. Can we make this optional with a flag? That can be set by SetIntervalPrint() or similar.

I would say that most of the time you are interested in the accumulated total time of each process, that is printed in PrintTimingInformation().